manta-util/src/collections/btree_map.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! An ordered map based on a B-Tree.
+
+use core::borrow::Borrow;
+
+#[doc(inline)]
+pub use alloc::collections::btree_map::*;
+
+/// Error Message for the [`pop_last`] Function
+const POP_LAST_ERROR_MESSAGE: &str =
+    "Querying the last element of the map must guarantee that `BTreeMap::remove` always succeeds.";
```

Comment:
Why are you writing this message separately instead of just plugging it in `pop_last`?

Reply:
This is a convention to put long and constant error message at the top of the code. If we have multiple such messages, it is easier to change without jumping to a specific line of code below and considering the code context.